### DISCUSSION TITLE
Dithering - Grid Artifacts

### DISCUSSION THREAD

**Discussion by aljazjelen** [arkypita/LaserGRBL#1512] [Q&A]
Hi everyone,

I am using Dithering Method (mostly Jarvis) and have had quite good results.

However, when I see from up close, every single picture has a grid-like artifacts. There are horizontal and vertical lines, which should not be there.

I tried different speeds and steps/min. I even tried different pictures and dithering methods... no success.
Please see below:

![IMG_20210706_193504](https://user-images.githubusercontent.com/18629679/124644176-42120200-de92-11eb-931b-3f9a03d2c583.jpg)
![IMG_20210706_193517](https://user-images.githubusercontent.com/18629679/124644181-43432f00-de92-11eb-87db-4c2203a84acf.jpg)

Has anyone encountered this before?

Thank you! :)

p.s. I have opened an Issue on the same topic. If that is not correct, please let me know, I will gladly close it. Sorry for the possible mess caused by this.

**Comment by gmmanonymus111** [comment]
Since I think it's more like a discussion than an issue with LaserGRBL, I'll comment here.
Have you checked similar issues that were previously reported? Please take a look at #30 and #1001, might help you out.
What kind of device do you have? Belt driven ones tend to do such pattern when the belt is too loose.

**Reply by aljazjelen** [reply]
Hi Alex, thanks for the comment.
Shall i close the issue then?

I have taken a look at #30 and I did a test for 9 different steps (starting at 9 per mm up to 18 per mm). All of them yield in the same result. I mean, not maybe exactly same grid pattern, but there was a grid pattern indeed. 

I might need to repeat, maybe extend up to more than 18 and less than 9.

On the other hand, I have not yet checked the firmware version of my GRBL, since the controller came "pre-flashed" from China.
But I think that i have 1.1 version, because S4 (Dynamic) is available. I dont know if it works tho.

I have a belt driven system with no lose belts. I can confirm this one.

Do you think, it would be worth checking something else? Maybe do the dithering out of LaserGRBL and just do the "Passthrough"?

Thannks!

**Reply by gmmanonymus111** [reply]
You could try passthrough if you have enough spare materials :)
Take a look at the steps/mm helper here:
![image](https://user-images.githubusercontent.com/9658243/124672358-2d943080-deb7-11eb-814f-bbd505e546e4.png)
This might help in setting a more easy to use steps/mm, which can reduce grids (because there will be no "rounded" steps)

**Reply by aljazjelen** [reply]
Hi Alex,

I have done exactly that, when I was doing steps per mm (9 to 18). But Didnt give me any good results. I was paying attention, that the number is round. Like in the ticker nmbr 30.

Do you think, that it might be worth changing the microstepping? 

Regards

**Reply by gmmanonymus111** [reply]
Well, I still think it's more like a hardware issue, so I don't think changing microstepping could help.
I don't know the direction you tried too engrave, but you could try the other way around (if you used vertical, try horizontal, or vice versa), maybe another dithering option, or even line to line? If grid is still there during line to line, it's definitely a hardware issue.
What kind of engraver do you have?

**Reply by aljazjelen** [reply]
Hi Alex,

I have Totem S 40W (5.5W Diode) engraver. I will try various directions (vertical, horizontal) and other Dithering option (I tried also Stucki, no success). I will try also line to line, just to check.

FYI, I tried Passthrough with a grayscale image. Well, the grid was still there. Altho not exactly the same proportion. I dont really understand how this is possible. What makes me more worried, I am really not a beginner in the CNC world... yet somehow I dont understand it. 

Just as you, I think its a HW issue. But not a mechanical. I would say its a problem of Stepper Motor control. It looks to me, that steps are not really followed properly. So there is constant error and when the error gets too big, it jumps a step, creating a line. I will try to engrave a parallelogram. If my theory is correct, then the "grid lines" should be skewed proportionally to the parallelogram. :)

Regards

**Reply by aljazjelen** [reply]
Hi Alex,

Did following tests:
- Horizontal vs Vertical vs Diagonal => Issues still there, gridlines present
- Different Dithering Methods => Issues still there, gridlines present
- Parallelogram Test => Issues still there, gridlines present

I am seriously losing hope :D 

Regards

**Comment by arkypita** [comment]
Assuming you have read carefully #30 and assuming you have a stepper resolution of 80 steps/mm reliable resolution are all multiples of your step size:


step | size (mm) | resolution (lines/mm)
-- | -- | --
1 | 0,0125 | 80
2 | 0,025 | 40
3 | 0,0375 | 26,66666667
4 | 0,05 | 20
5 | 0,0625 | 16
6 | 0,075 | 13,33333333
7 | 0,0875 | 11,42857143
8 | 0,1 | 10
9 | 0,1125 | 8,888888889
10 | 0,125 | 8
11 | 0,1375 | 7,272727273
12 | 0,15 | 6,666666667
13 | 0,1625 | 6,153846154
14 | 0,175 | 5,714285714
15 | 0,1875 | 5,333333333
16 | 0,2 | 5
17 | 0,2125 | 4,705882353
18 | 0,225 | 4,444444444
19 | 0,2375 | 4,210526316
20 | 0,25 | 4
21 | 0,2625 | 3,80952381
22 | 0,275 | 3,636363636
23 | 0,2875 | 3,47826087
24 | 0,3 | 3,333333333
25 | 0,3125 | 3,2
26 | 0,325 | 3,076923077
27 | 0,3375 | 2,962962963
28 | 0,35 | 2,857142857
29 | 0,3625 | 2,75862069
30 | 0,375 | 2,666666667
31 | 0,3875 | 2,580645161
32 | 0,4 | 2,5
33 | 0,4125 | 2,424242424
34 | 0,425 | 2,352941176
35 | 0,4375 | 2,285714286
36 | 0,45 | 2,222222222
37 | 0,4625 | 2,162162162
38 | 0,475 | 2,105263158
39 | 0,4875 | 2,051282051
40 | 0,5 | 2
41 | 0,5125 | 1,951219512
42 | 0,525 | 1,904761905

Within these values: 80, 40, 20, **16, 10, 8, 5, 4**, 3.2, 2.5, 2 are the values of lines/mm I suggest you to use.

**Reply by arkypita** [reply]
be aware that this could be a completely hardware problem. if there is friction / resistance in the sliding of the carriage, the belt (which has a minimum of elasticity) stretches but the carriage moves little (less than it should) until the tension on the belt reaches a point where the carriage makes a wider movement in which he recovers the excessive tension

**Reply by aljazjelen** [reply]
Hi Arkyipita,

I have dared to try everything to my power with current HW. Indeed the steps limitation is known to me from my other CNC projects. For this reason I focused myself on the values you mentioned (I also tried 20 steps/mm).

Somehow I dont get a proper results. Grid artifacts everywhere.

The funny thing tho, I dared to try Light Burn as well. Results between the two software are not exactly the same. The same picture input... BUT the biggest difference is at 20 steps/mm. At that value, the Light Burn and LaserGRBL perform very differently. I will try again today and make side by side comparison.

Best regards

**Reply by arkypita** [reply]
> The funny thing tho, I dared to try Light Burn as well. Results between the two software are not exactly the same. The same picture input... BUT the biggest difference is at 20 steps/mm. At that value, the Light Burn and LaserGRBL perform very differently. I will try again today and make side by side comparison.

I've never used lightburn so I can't tell you, anyway under each program there is the generated gcode: different the gcode different are the results. 

You can try to see what the gcode looks like and compare it with a simple notepad program. In lasergrbl you can save the gcode from the file menu, in lightburn there will be something similar.

If you open a gcode file generated by LaserGRBL with 8 lines / mm you will find that all the X and Y commands will be at multiple positions of 1/8

```
G0 X0 Y0 F4000
M3 S0
G0 X0.25 Y0 S0
G1 X0.375 S1000
G0 X0.75 Y0 S0
G1 X0.875 S1000
G0 X1.25 Y0 S0
G1 X1.375 S1000
G0 X1.75 Y0 S0
[...]
G0 X99.75 Y0.125 S0
G1 X99.625 S1000
G0 X99.25 Y0.125 S0
G1 X99.125 S1000
G0 X99 Y0.125 S0
G1 X98.875 S1000
G0 X98.625 Y0.125 S0
G1 X98.5 S1000
G0 X98.25 Y0.125 S0
G1 X98.125 S1000
G0 X97.875 Y0.125 S0
```
